examples/basic-usage.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+require __DIR__ . '/../vendor/autoload.php';
+
+use ArchiPro\EventDispatcher\AsyncEventDispatcher;
+use ArchiPro\EventDispatcher\Event\AbstractEvent;
+use ArchiPro\EventDispatcher\ListenerProvider;
+use Revolt\EventLoop;
+
+// Create a custom event
+class UserCreatedEvent extends AbstractEvent
+{
+    public function __construct(
+        public readonly string $userId,
+        public readonly string $email
+    ) {
+    }
+}
+
+// Create the listener provider and register listeners
+$listenerProvider = new ListenerProvider();
+
+$listenerProvider->addListener(UserCreatedEvent::class, function (UserCreatedEvent $event) {
+    // Simulate async operation
+    EventLoop::delay(1, fn () => null);
+    echo "Sending welcome email to {$event->email}\n";
```

Comment:
This echo is meant to be within the delay callback I believe. or do you mean to use the delay() function to [async] block?